channel/state/outcome/allocation.go

```diff
@@ -105,3 +106,51 @@ var allocationsTy = abi.ArgumentMarshaling{
 		{Name: "metadata", Type: "bytes"},
 	},
 }
+
+// DivertToGuarantee returns a new Allocations, identical to the reciever but with
+// the leftDebtee's amount reduced by leftDebit,
+// the rightDebtee's amount reduced by rightDebit,
+// and a Guarantee appended for the guaranteeDestination
+func (allocations Allocations) DivertToGuarantee(
+	leftDestination types.Destination,
+	rightDestination types.Destination,
+	leftAmount *big.Int,
+	rightAmount *big.Int,
+	guaranteeDestination types.Destination,
+) (Allocations, error) {
+
+	if leftDestination == rightDestination {
+		return Allocations{}, errors.New(`debtees must be distinct`)
+	}
+
+	newAllocations := make([]Allocation, len(allocations)+1)
```

Comment:
```suggestion
	newAllocations := make([]Allocation, len(allocations), len(allocations)+1)
```
Doesn't place a blank `Allocation{}` in the last slot - a requirement of the suggestion on line 148.

I'm not sure whether
```
newAllocations := make([]Allocation, 0, len(allocations)+1)
// followed by
newAllocations = append(newAllocations, allocation) // inside the loop, instead of `newA[i] = allocations`
```
is better still.

Reply:
Yes I went with you "better still" suggestion. Nice use of a zero length, nonzero capacity slice.